src/rt-app.c

```diff
@@ -108,6 +108,64 @@ static void thread_data_set_unique_name(thread_data_t *tdata, int nforks)
 	}
 }
 
+/*
+ * Create a new pthread using the info provided in the passed @td thread_data_t.
+ *
+ * @td:		The thread data of the task to create - we copy it and leave
+ *		it unmodified for other fork events to use to create additional
+ *		threads based on the same tdata.
+ *
+ * @index:	Index of the task to create.
+ *
+ * @forked:	Whether this task is a craeted from fork event or at
+ *		application startup.
+ *
+ * @nforks:	If this is a forked task, we use nforks to give it a unique name.
+ *
+ * Returns 0 on success or -1 on failure.
+ */
+static int create_thread(const thread_data_t *td, int index, int forked, int nforks)
+{
+	thread_data_t *tdata;
+
+	if (!td) {
+		log_error("Failed to fork new thread, passed NULL thread_data_t: %s", td->name);
+		return -1;
+	}
+
+	tdata = malloc(sizeof(thread_data_t));
+	if (!tdata) {
+		log_error("Failed to allocate memory for a new fork: %s", td->name);
+		return -1;
```

Comment:
It's not only for fork event so I prefer to keep the generic log : "Failed to duplicate thread data: %s", td->name

That's the only comment i have. Otherwise looks good to me

Reply:
Indeed. Copy paste error from my side. Fixed it locally and will hammer it today and tomorrow and push the final branch for merging on Friday.

Thanks a lot for you time!